### ISSUE TITLE
[keymgr] D2 Signoff

### ISSUE THREAD

**Issue by msfschaffner** [lowRISC/opentitan#20981]
### Description

Ensure D2 signoff criteria are fulfilled after focus area changes have landed.

**Comment by andreaskurth** [comment]
**This analysis is work in progress.**

## Commits since Earlgrey-ES tapeout

```
git rev-parse HEAD
```
c73929e4d6e864bad72c456b265a089225a5c8e2

```
git log Earlgrey-M2.5.2-RC0..HEAD --oneline hw/ip/keymgr
```
- 52da8f707d [keymgr] Add missing CM annotation
  --> RTL changed but only a CM label added
- 0d14dd693f [docs] Fix repetitions of the definite article
  --> no RTL changed; spelling fixed in spec
- 56399241e7 Revert "[edn] Move prim_edn_req out of prim"
  --> reverts 3b4e36e01c below
- c721c51c13 [rtl, prim] Add 'commit' functionality to prim_count
  --> RTL changed but no functional difference
- e81b5885e5 [keymgr/otp_ctrl] Add support for creator/owner seeds
  --> RTL change with functional impact; to be reviewed in detail
- 61a237e197 [util/reggen] reverse order of substruct generation
  --> RTL changed but no functional difference
- 3b4e36e01c [edn] Move prim_edn_req out of prim
  --> reverted by 56399241e7 above
- de31bdf1c2 [reggen] Remove the devmode input
    --> RTL changed but no functional difference
- 963a5006cc [doc] Minor tweak to md sanitisation code
  --> no RTL changed; spec docs changed but changes should be structural only
- 975a6eb927 [adc_ctrl,dv] Tidy up access to intr_state in env_cfg files
  --> no RTL nor specification changed  
- 1b16ca2122 [reggen] Add mubi support SWAccess that sets/clears a reg
  --> RTL changed but no functional difference
- 59f8142826 [doc] Moved badges over to using hosted images
  --> no RTL nor specification changed
- 8ba7e45f4e [SiVal] Test plan updates for keymgr
  --> no RTL nor specification changed
- 19c05ad821 [doc] keymgr registers and interfaces now use CMDGEN
  --> no RTL changed; spec docs changed but changes should be structural only
- 7688e714e8 [reggen] Add initial support for version and cip_id hjson fields
  --> no RTL changed
- fbd888eea8 Revert "[reggen] Add CIP_IDs and bump all major versions"
  --> reverts commit below
- 0ba10b3cd3 [reggen] Add CIP_IDs and bump all major versions
  --> reverted by commit above
- e47df29f3e [misc] Use lc_tx_t testing functions at endpoints
  --> RTL changed but no functional difference

## Issues closed since the Earlgrey-ES tapeout

## Currently open issues

## Coverage report from <DD/MM/YYYY>

## Summary

**Comment by andreaskurth** [comment]
@vogelpi: May I ask you for feedback on my analysis and D2 signoff approval if you agree?

**Comment by vogelpi** [comment]
Thanks for putting this together @andreaskurth !

To be honest, I am in favor of signing of D2S. This is the state keymgr was before and as you pointed out, no significant changes have gone in since the last sign-off. So there is IMO no need to go back down again. I agree that there are a couple of security hardening changes we want to do for M3. But all of them are minor. We also keep adding security improvements to other blocks after hitting D2S.

D2S just says security countermeasures are implemented and this is the case for keymgr. The things coming in M3 are not about adding new countermeasures but minor improvements to existing things.

So to summarize: signing of at D2 is definitely okay, signing of at D2S would be preferred from my side. It will save us further paper work in M3.

**Comment by andreaskurth** [comment]
Alright, given that (1) keymgr has been signed off at D2S before and the changes since then don't revert D2S and (2) #8120 is tracked for M3, I agree that we can sign keymgr off at D2S again.
